### PR TITLE
Add `githubscm.updateGithubCommitStatus` methods

### PR DESCRIPTION
**Pull request by radtriste**

### CODE REVIEW

**Comment by radtriste** [comment]
jenkins test this

**Review by lampajr — APPROVED** [review]
LGTM

File: vars/githubscm.groovy [hunk 1]
```diff
@@ -499,3 +506,91 @@ def getLatestTag(String startsWith = '', String endsWith = '', List ignoreTags =
...
+    println "[INFO] Update commit status for check ${checkName}: state = ${state} and message = ${message}"
+    if (!getCommitStatusRepoURLEnv() || !getCommitStatusShaEnv()) {
+        println '[INFO] Commit status info are not stored, guessing from current repository'
+        setCommitStatusRepoURLEnv()
+        setCommitStatusShaEnv()
+    }
+    println "[DEBUG] repo url = ${getCommitStatusRepoURLEnv()}"
+    println "[DEBUG] commit sha = ${getCommitStatusShaEnv()}"
+
+    step([
```
Comment (Ginxo) [inline vars/githubscm.groovy:557]: I love it! :heart:

File: vars/githubscm.groovy [hunk 2]
```diff
@@ -499,3 +506,91 @@ def getLatestTag(String startsWith = '', String endsWith = '', List ignoreTags =
...
+    String buildResult = currentBuild.currentResult
+    println "[DEBUG] Got build result ${buildResult}"
+
+    String testsInfo = ''
+    def testResults = util.retrieveTestResults()
+    if(testResults) {
+        int pass = testResults.passCount
+        int skip = testResults.skipCount
+        int fail = testResults.failCount
+        additionalInfo = "${pass + skip + fail} tests run, ${failCount} failed, ${skipCount} skipped."
```
Comment (Ginxo) [inline vars/githubscm.groovy:577]: shouldn't it be `testsInfo`? :thinking: something is not matching here FMPOV... :thinking: 
```suggestion
        testsInfo = "${pass + skip + fail} tests run, ${failCount} failed, ${skipCount} skipped."
```

File: vars/githubscm.groovy [hunk 3]
```diff
@@ -499,3 +506,91 @@ def getLatestTag(String startsWith = '', String endsWith = '', List ignoreTags =
...
+    }
```
Comment (Ginxo) [inline vars/githubscm.groovy:578]: another recommendation would be...
```suggestion
    String testsInfo = testResults ? "${testResults.passCount + testResults.skipCount + testResults.failCount} tests run, ${failCount} failed, ${skipCount} skipped." : ''
    def testResults = util.retrieveTestResults()
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
please check my comments, specially the ones from `githubscm.updateGithubCommitStatusFromBuildResult`, the other one from `githubscm.prepareCommitStatusInformation` is just a recommendation

File: vars/githubscm.groovy (hunk 2 above)
Comment (radtriste) [reply vars/githubscm.groovy:577]: yeah indeed. And yesterday evening I could not see what was wrong in the tests ^^ Gonna correct this. Thanks for catching it !

File: vars/githubscm.groovy [hunk 4]
```diff
@@ -499,3 +506,91 @@ def getLatestTag(String startsWith = '', String endsWith = '', List ignoreTags =
     cmd += ' | head -n 1'
     return sh(returnStdout: true, script: cmd).trim()
 }
+
+/*
+* Store in env the commit info needed to update the commit status
+*/
+void prepareCommitStatusInformation(String repository, String author, String branch, String credentialsId = 'kie-ci') {
+    String tempDir = util.generateTempFolder()
+    dir(tempDir) {
```
[the remark this answers is not in the document]
Comment (radtriste) [reply vars/githubscm.groovy:515]: added

File: vars/githubscm.groovy (hunk 2 above)
Comment (radtriste) [reply vars/githubscm.groovy:577]: corrected

File: vars/githubscm.groovy (hunk 3 above)
Comment (radtriste) [reply vars/githubscm.groovy:578]: updated

**Review by Ginxo — APPROVED** [review]
LGTM, many thanks

File: vars/githubscm.groovy [hunk 5]
```diff
@@ -459,12 +470,7 @@ def updateReleaseBody(String tagName, String credsId = 'kie-ci') {
             sed -i -r 's|(KOGITO-[0-9]*)(.*)|\\[\\1\\](https\\://issues\\.redhat\\.com/browse/\\1)\\2|g' ${releaseNotesFile}
 
             sed -i -r 's|\\[(DROOLS-[0-9]*)\\](.*)|\\1\\2|g' ${releaseNotesFile}
-            sed -i -r 's|(DROOLS-[0-9]*)(.*)|\\[\\1\\](https\\://issues\\.redhat\\.com/browse/\\1)\\2|g' ${releaseNotesFile}
-
-            sed -i -r 's|\\[(BXMSPROD-[0-9]*)\\](.*)|\\1\\2|g' ${releaseNotesFile}
-            sed -i -r 's|(BXMSPROD-[0-9]*)(.*)|\\[\\1\\](https\\://issues\\.redhat\\.com/browse/\\1)\\2|g' ${releaseNotesFile}
-
-            sed -i -r 's|\\[(kie-issues-[0-9]*)\\](.*)|\\1\\2|g' ${releaseNotesFile}
+            sed -i -r 's|(DROOLS-[getGitRepositoryNamesues-[0-9]*)\\](.*)|\\1\\2|g' ${releaseNotesFile}
```
Comment (radtriste) [inline vars/githubscm.groovy:473]:
```suggestion
            sed -i -r 's|(DROOLS-[0-9]*)(.*)|\\[\\1\\](https\\://issues\\.redhat\\.com/browse/\\1)\\2|g' ${releaseNotesFile}
            sed -i -r 's|\\[(BXMSPROD-[0-9]*)\\](.*)|\\1\\2|g' ${releaseNotesFile}
            sed -i -r 's|(BXMSPROD-[0-9]*)(.*)|\\[\\1\\](https\\://issues\\.redhat\\.com/browse/\\1)\\2|g' ${releaseNotesFile}
            sed -i -r 's|\\[(kie-issues-[0-9]*)\\](.*)|\\1\\2|g' ${releaseNotesFile}
```

**Pull request merged**